bin/main.ml

```diff
@@ -2,8 +2,13 @@ open Olinkcheck
 
 let file_contents file = In_channel.(with_open_bin file In_channel.input_all)
 
+let pretty_print_link_status (link, (status_code, status_string)) =
+  Printf.printf "%s - %i %s\n" link status_code status_string
+
 let () =
   if Array.length Sys.argv < 2 then
     print_endline "Usage: olinkcheck filename.md"
   else
-    List.iter print_endline (Parser.extract_links (file_contents Sys.argv.(1)))
+    let links = Parser.extract_links (file_contents Sys.argv.(1)) in
+    let statuses = List.map Link.get_link_status links in
+    List.iter pretty_print_link_status (List.combine links statuses)
```

Comment:
Here, you have two loops, back to back (`List.map` and `List.iter`). You should take advantage of the fundamental property of `map` (where `x |> f` is simply `f x`): 
```ocaml
fun u -> u |> map f |> map g = map (fun x -> x |> f |> g)
```
This will allow you to have a single loop